x/claim/spec/README.md

```diff
@@ -0,0 +1,67 @@
+# `claim`
+
+## Abstract
+
+This specifies the `claim` module of Osmosis project, provide commands for claimable amount query and claim airdrop.
+We apply real-time decay after `DurationUntilDecay` pass where monthly decay rate is `-10%` of inital airdrop amount.
+When `DurationUntilDecay + DurationOfDecay` time passes, all unclaimed coins will be sent to the community pool.
+
+## Genesis State
+
+### Accounts
+
+All genesis accounts have `1 Osmo` for claim fee.
+
+### Claimables
+
+Claimables are the maximum claimable amounts per address and it's configured by genesis.
+It's determined by applying few rules to the snapshot of ATOM balance at cosmoshub-3.
+
+### User actions
+User need to accomplish `DelegateStake`, `Vote`, `AddLiquidity`, `Swap` actions to completely withdraw his claimables.
+Each action is defined in enum in implementation
+```
+	ActionAddLiquidity  Action = 0
+	ActionSwap          Action = 1
+	ActionVote          Action = 2
+	ActionDelegateStake Action = 3
+```
+All of these actions are monitored by registring claim **hooks** to governance, staking, gamm, lockup modules hooks.
+
+### User withdrawables
+It is how much a user can withraw now.
+```
+user_withdrawables = user_claimables x (percentage_accomplished_actions - percentage_withdrawn_actions)
+```
+
+### User withdrawn actions
+A user can't withdraw twice for single action type, example `Vote`. To withdraw all, they need to do every action.
+To do this, claim module manage withdrawn actions flag. It's using same strategy as user actions, but only count the actions that user has already withdrawn.
+If Alice has done `DelegateStake` action, withdraw 25% of claimables, and then if she do `Vote` action, she is able to withdraw another 25% of claimables. Here, withdrawn actions flag is used to avoid double withdraw on `DelegateStake` action.
+
+### Airdrop Tools
+There are tools to generate genesis from cosmos-hub snapshot.
+
+#### Genesis generation
+
+Generate genesis from cosmos-hub snapshot genesis and output snapshot of atom, osmo balance and percentage by address.
+```sh
+osmosisd export-airdrop-genesis uatom ../genesis.json --total-amount=100000000000000 --snapshot-output="../snapshot.json"
+osmosisd export-airdrop-genesis uatom ../genesis.json --snapshot-output="../snapshot.json"
+```
+
+## Queries
+
+Query claimable amount for a given address at the current time.
+```sh
+osmosisd query claim claimable $(osmosisd keys show -a validator --keyring-backend=test)
+```
+
+## Msgs
```

Comment:
Why do people have to manually send a claim msg?  Why don't we just transfer the coins during the hook itself (when the flag is set)?

Reply:
It would be very good to do it within hook, manual interaction is making codebase complex

Reply:
Let's go ahead and do that.  Yeah, seems like it would get rid of a lot of the complexity.

Reply:
Yes, it's done, can you check again?